### DISCUSSION TITLE
Licensing

### DISCUSSION THREAD

**Discussion by OrigamiPT** [IgniteUI/app-builder#6] [Ideas]
Hello,

i have an ideia to share with you all, i just purchased a plan for the igniteUI and i saw that when i pull the generated code from indigo that the licensing is not applied and with that each time someone pulls the code from there it has to apply all the licensing stuff to the code, even if i create just a new area i have to change all the code to reflect the licensing.

So the ideia would be for the code generator to check if a license is in place and apply it to the code.

Best regard.

**Comment by kdinev** [comment]
@damyanpetev @zdrawku @pmoleri Guys, we should output the generated code with the licensed version of the packages, instead of the trial ones, when a licensed users is initiating the code generation.

**Comment by damyanpetev** [comment]
I was going to mention upgrading the package should be a simple [single command affair](https://www.infragistics.com/products/ignite-ui-angular/angular/components/general/ignite-ui-licensing#upgrading-packages-using-our-angular-schematics-or-ignite-ui-cli) but then I read this bit:
> [...] even if i create just a new area i have to change all the code to reflect the licensing [...]

So assuming you made the licensing changes already and a new generation comes with the entire project overriding them, I can see how that can be a hassle. Are you perchance publishing to GitHub repo? Because the overriding behavior is on our radar and we might have some ideas for GitHub PRs that'll possibly help alleviate this as well.

**Reply by OrigamiPT** [reply]
In my case i am not using a GitHub

**Comment by zdrawku** [comment]
@OrigamiPT we have some ideas on how to tacke this problem. The work item is part of our next sprint iteration. I will update the disucssion when we are ready with the implementation.

**Comment by zdrawku** [comment]
@OrigamiPT with today's release of App Builder you will get a licensed code upon export for licensed accounts and code with watermark for trial users. 

1. If you are a licensed user, then the licensed packages of [Ignite UI Angular](https://www.infragistics.com/products/ignite-ui-angular) will be used, upon app generation. This applies for both application download and publish to GitHub. When the application is published to GitHub we add a CI that will build your project and run basic tests. We also publish NPM_AUTH_TOKEN that is required from the GitHub CI, in order to be able to use the licensed package.

2. If you are a trial user then the free trial package of [Ignite UI Angular](https://www.infragistics.com/products/ignite-ui-angular) will be used, upon app generation. This applies for both application download and publish to GitHub.


Thank you for making AB better!
https://www.infragistics.com/community/blogs/b/infragistics/posts/app-builder-release-with-tree-grid-and-more-components
